ckan/config/middleware/flask_app.py

```diff
@@ -71,6 +73,27 @@ def make_flask_stack(conf, **app_conf):
         app.config['DEBUG_TB_INTERCEPT_REDIRECTS'] = False
         DebugToolbarExtension(app)
 
+    # Use Beaker as the Flask session interface
+    class BeakerSessionInterface(SessionInterface):
+        def open_session(self, app, request):
+            if 'beaker.session' in request.environ:
+                return request.environ['beaker.session']
+
+        def save_session(self, app, session, response):
+            session.save()
+
+    cache_dir = app_conf.get('cache_dir') or app_conf.get('cache.dir')
+    session_opts = {
+        'session.data_dir': '{data_dir}/sessions'.format(
```

Comment:
Do we *really* want to use the file-based session type? We could do `'session.type': 'ext:database'` and store it in postgres.